From 6928f9fe30205043b1bce43d1fd3aa69c057e822 Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Thu, 19 Nov 2020 17:12:03 +0300
Subject: [PATCH 15/18] media: i2c: imx219: Set subdevice fwnode

This initializes subdevice fwnode since it's used
by the V4L2 device OF endpoint parser, for example,
R-Car CSI2 receiver driver.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/media/i2c/imx219.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/i2c/imx219.c b/drivers/media/i2c/imx219.c
index 4771d0ef2c46..c55fd733fd80 100644
--- a/drivers/media/i2c/imx219.c
+++ b/drivers/media/i2c/imx219.c
@@ -1510,6 +1510,7 @@ static int imx219_probe(struct i2c_client *client)
 	imx219->sd.internal_ops = &imx219_internal_ops;
 	imx219->sd.flags |= V4L2_SUBDEV_FL_HAS_DEVNODE;
 	imx219->sd.entity.function = MEDIA_ENT_F_CAM_SENSOR;
+	imx219->sd.fwnode = fwnode_graph_get_next_endpoint(dev_fwnode(dev), NULL);
 
 	/* Initialize source pad */
 	imx219->pad.flags = MEDIA_PAD_FL_SOURCE;
-- 
2.25.1

